lib/services/api/api_service.ts

```diff
@@ -234,4 +256,56 @@ export class SNApiService extends PureService {
 
     return response;
   }
+
+  private async checkForExpiredAccessToken(errorResponse: HttpResponse) {
+    if (errorResponse.status !== EXPIRED_ACCESS_TOKEN_RESPONSE_STATUS) {
+      return;
+    }
+    await this.refreshSession();
+  }
+
+  private async refreshSession() {
+    if (this.refreshingSession) {
+      return;
+    }
+    const user = await this.storageService!.getValue(StorageKey.User);
+    if (!user) {
+      return;
+    }
+    if (!user.uuid) {
+      return;
+    }
+    const userUuid = user.uuid;
+    this.refreshingSession = true;
+    const url = await this.path(REQUEST_PATH_SESSION_REFRESH);
+    const params = this.params({
+      user_uuid: userUuid,
```

Comment:
Why does the server need a user_uuid? Can't it determine the user based on the access_token or refresh_token?